src/BTree.mo

```diff
@@ -57,14 +58,37 @@ module {
     };
   };
 
+  public module Insert {
+
+
+
+  };
+
+  // Assert that the given B-Tree instance observes all relevant invariants.
+  // Used for unit tests.  Show function helps debug failing tests.
+  //
+  // Note: These checks-as-assertions can be refactored into value-producing checks,
+  // if that seems useful.  Then, they can be individual matchers tests.  Again, if useful.
+  public func assertIsValid<K, V>(
+    t : Tree<K, V>,
+    compare : (K, K) -> Order.Order,
+    show : K -> Text)
+  {
+    Check.root<K, V>({compare; show}, t)
+  };
+
+  public func assertIsValidTextKeys<V>(t : Tree<Text, V>){
+    Check.root<Text, V>({compare=Text.compare; show=func (t:Text) : Text { t }}, t)
+  };
```

Comment:
Would it help if `Check` and these test methods were just moved into `test/BTreeTest.mo`?